mailbox/opensearch/src/main/java/org/apache/james/mailbox/opensearch/events/OpenSearchListeningMessageSearchIndex.java

```diff
@@ -273,8 +272,11 @@ private ArrayNode extractUserFlags(ObjectNode source) {
     private void extractMessageIdFromHit(Hit<ObjectNode> hit, SynchronousSink<MessageId> sink) {
         JsonData messageId = hit.fields().get(MESSAGE_ID);
         if (messageId != null) {
-            List<String> extractMessageId = messageId.deserialize(JsonpDeserializer.arrayDeserializer(JsonpDeserializer.stringDeserializer()));
-            sink.next(messageIdFactory.fromString(extractMessageId.get(0)));
```

Comment:
Q: How did you diagnose that `JsonpDeserializer.arrayDeserializer seems to do some blocking action in its code...`?

Reply:
jstack and flame graphs during perf tests, where you could see there was an issue around there in the stack traces